dialogue-clients/src/main/java/com/palantir/dialogue/clients/ServicesConfigBlockWithResolvedHosts.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.clients;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.logsafe.DoNotLog;
+import java.net.InetAddress;
+import org.immutables.value.Value;
+
+@DoNotLog
+@Value.Immutable
+interface ServicesConfigBlockWithResolvedHosts {
+    @Value.Parameter
+    ServicesConfigBlock scb();
```

Comment:
@carterkozak thoughts on making this an `Optional`? a thing i'm concerned about is: we ideally need to set an initial value for a `SettableRefreshable` of this type (see here: https://github.com/palantir/dialogue/pull/2141/files#diff-6d1086f7aca5ef649647a82c52683f1d85aa8c3a9568e9b4510669f1e749fc89R75-R76) which will be provided before the full round of initial name resolutions has completed. We could construct an "empty" `ServicesConfigBlock` but i worry that becomes a bit error-prone if the types within change to being required at build time.

Reply:
If those types changed, we’d break a lot more than this — I don’t think it would be possible to add required types to that change.